python_modules/dagster/dagster/_core/definitions/multi_dimensional_partitions.py

```diff
@@ -50,3 +60,123 @@ def __new__(cls, partition_dimension_mapping: Mapping[str, str]):
 
     def keys_by_dimension(self):
         return {dim_key.dimension_name: dim_key.partition_key for dim_key in self.dimension_keys}
+
+
+class PartitionDimensionDefinition(
+    NamedTuple(
+        "_PartitionDimensionDefinition",
+        [
+            ("name", str),
+            ("partitions_def", PartitionsDefinition),
+        ],
+    )
+):
+    def __new__(
+        cls,
+        name: str,
+        partitions_def: PartitionsDefinition,
+    ):
+        return super().__new__(
+            cls,
+            name=check.str_param(name, "name"),
+            partitions_def=check.inst_param(partitions_def, "partitions_def", PartitionsDefinition),
+        )
+
+
+@experimental
+class MultiPartitionsDefinition(PartitionsDefinition):
+    """The set of partitions is the cross product of partitions in the inner partitions
+    definitions"""
```

Comment:
It would be helpful to add an attributes section of the docstring that explains what the keys and values of the  partitions_defs attr mean.